add user-agent to client

dandye opened the pull request.

emeryray2002 commented:
I think this would overwrite the AuthorizedSession object?
Maybe like self._session.headers["User-Agent"] = USER_AGENT ?

dandye commented:
Thanks for the review\! I've addressed your feedback:

- Fixed the bug where `self._session` was being overwritten with the USER_AGENT string
- Moved the User-Agent header setting to the `__init__` method so it only happens once during initialization
- Added support for custom User-Agent via `SECOPS_WRAPPER_USER_AGENT` environment variable
- The implementation now preserves any existing User-Agent header if one is already set

The latest commit includes all these improvements.
